### PR TITLE
feat: Adjust tabs to fiori3 specification

### PR DESCRIPTION
**Pull request by JKMarkowski**
## Related Issue
Closes SAP/fundamental-styles#454

## Description
There is new styling for tabs + new options, which follows latest fiori3 requirements.

## Brreaking Changes
* No more `disabled` state for tabs (if it's disabled, it shouldn't be visible)
* Compact mode is added to tabs `fd-tabs--compact`
* Semantic mode added to tabs `fd-tabs__item--` + `success / warning / information / error / neutral` 
* Filter / Process / Icon Only modes added to tabs.
### **Screenshots will be added soon**

> **NOTE:** If you've made any style changes, please provide appropriate screenshots (before and after) to help reviewers.
>
> To see examples of which screenshots to include, go to [Screenshot Examples](https://github.com/SAP/fundamental-styles/wiki/Pull-Request-Screenshot-Examples).

### After:
Basic:
image

Basic with counter:
image

Icon Only Cozy:
image

Icon Only Compact:
image

Progress Cozy:
image

Progress Compact:
image

Filter Cozy:
image

Filter Compact:
image

Semantic Icons only:
image

Semantic Inline
image

### CODE REVIEW

File: src/tabs.scss [hunk 1]
```diff
@@ -12,16 +12,56 @@
...
-  $fd-tabs-link-padding-y: fd-space("s") !default;
+  $fd-tabs-selected-color: var(--sapSelectedColor);
+  $fd-tabs-link-padding-y: 0.875rem !default;
+  $fd-tabs-icon-link-padding-y: 1rem !default;
   $fd-tabs-link-padding-x: fd-space("s") !default;
   $fd-tabs-link-border-width: 3px !default;
   $fd-tabs-link-border-color: fd-color("action", 1) !default;
   $fd-tabs-link-transition-params: $fd-animation--speed ease-in !default;
+  $fd-tabs-icon-size: 1.125rem;
+  $fd-tabs-icon-selection-height: 2.625rem;
```
Comment (stefanoScalzo) [inline src/tabs.scss]: should be 2.75rem per speck

Comment (JKMarkowski) [reply src/tabs.scss]: it's `2.75rem`, but including border, which is `1px` per side

File: test/templates/tabs/index.njk [hunk 2]
```diff
@@ -51,6 +51,423 @@
...
+    </li>
+    <li class="fd-tabs__item fd-tabs__item-">
+        <a class="fd-tabs__link" aria-controls="QrQ5Cl" href="#QrQ5Cl" role="tab">
+            <span class="fd-tabs__icon sap-icon--cart">
+                <p class="fd-tabs__count">1</p>
+            </span>
+        </a>
+    </li>
+    <li class="fd-tabs__item fd-tabs__item--icon-only">
+        <a class="fd-tabs__link" aria-controls="t8oU9X" aria-disabled="true" role="tab">
```
Comment (stefanoScalzo) [inline test/templates/tabs/index.njk]: remove disabled examples

File: src/tabs.scss [hunk 3]
```diff
@@ -12,16 +12,56 @@
 $block: #{$fd-namespace}-tabs;
 
 .#{$block} {
+  @mixin tabs-semantic($color) {
+    .#{$block}__icon {
```
Comment (stefanoScalzo) [inline src/tabs.scss]: height should be 2.75rem

Comment (stefanoScalzo) [reply src/tabs.scss]: And icon height should be 1.25rem

File: src/tabs.scss [hunk 4]
```diff
@@ -41,24 +81,72 @@ $block: #{$fd-namespace}-tabs;
...
   }
 
   &__link {
     @include fd-reset();
 
     display: block;
     position: relative;
-    padding: $fd-tabs-link-padding-y 0;
+    padding: 0 $fd-tabs-selection-padding-x;
     text-decoration: none;
```
Comment (stefanoScalzo) [inline src/tabs.scss]: check for overflow. Might need to contact the designers

File: src/tabs.scss [hunk 5]
```diff
@@ -107,4 +194,165 @@ $block: #{$fd-namespace}-tabs;
...
+  &--filter {
+    display: flex;
+    align-items: center;
+
+    .#{$block}__item {
+      padding: 0 0.5rem;
+    }
+
+    .#{$block}__icon {
+      margin-bottom: 0.375rem;
```
Comment (stefanoScalzo) [inline src/tabs.scss]: check for text overflow. Might need to contact designers

File: src/tabs.scss [hunk 6]
```diff
@@ -41,24 +81,72 @@ $block: #{$fd-namespace}-tabs;
...
+    &--neutral {
+      @include tabs-semantic(var(--sapNeutralColor));
+    }
+  }
+
+  &__icon {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    width: $fd-tabs-icon-selection-height;
```
Comment (stefanoScalzo) [inline src/tabs.scss]: should have min-width of 2rem

File: src/tabs.scss (hunk 3 above)
Comment (JKMarkowski) [reply src/tabs.scss]: It's the same as above. Height is 2.625rem + 0.125rem(from border)

File: src/tabs.scss (hunk 6 above)
Comment (JKMarkowski) [reply src/tabs.scss]: Selectiion size of icon 
Cozy mode - 2.625rem height/width + (2 * 0.625rem) from borders
Compact mode - 1.875rem height/width + (2 * 0.625rem) from borders
General size matches fiori3, Also checked with https://sapui5.hana.ondemand.com/#/entity/sap.m.IconTabBar

**Comment by InnaAtanasova** [comment]
Screen Shot 2019-12-23 at 12 19 35 PM

Not using the correct font-size and color

**Comment by InnaAtanasova** [comment]
The selection bar specs are probably updated because there are some differences:
- the height should be 0.188rem
- border-radius should be 0.125rem
- the colors should be updated

**Comment by InnaAtanasova** [comment]
Missing container top and bottom borders

**Comment by InnaAtanasova** [comment]
Screen Shot 2019-12-23 at 12 33 52 PM

Focus color is not correct. 

Also, here https://sapui5.hana.ondemand.com/#/entity/sap.m.IconTabBar/sample/sap.m.sample.IconTabBarFiori2 they only focus the text, not the counter. Should ask the designers about the behaviour.

**Comment by InnaAtanasova** [comment]
Screen Shot 2019-12-23 at 12 36 56 PM

Icon size is 1.25rem. Should result in 20px

**Comment by InnaAtanasova** [comment]
The alignment of the item counter seems a bit off
Screen Shot 2019-12-23 at 12 47 44 PM

This is how it looks in the demo:
Screen Shot 2019-12-23 at 12 47 38 PM

**Comment by InnaAtanasova** [comment]
Screen Shot 2019-12-23 at 12 50 47 PM
The icon of the overflow button is wrong

File: src/tabs.scss [hunk 7]
```diff
@@ -107,4 +195,171 @@ $block: #{$fd-namespace}-tabs;
...
+
+  &__counter-header {
+    font-size: 1.5rem;
+    line-height: 1.5rem;
+    color: $fd-tabs-label-color;
+    padding-right: 0.5rem;
+  }
+
+  &__process {
+    padding: 0 0.375rem 0.25rem 0.25rem;
```
Comment (InnaAtanasova) [inline src/tabs.scss]: I am not sure these values are correct.

**Comment by JKMarkowski** [comment]
Thank you @InnaAtanasova for the review, I will put the changes there

**Comment by stefanoScalzo** [comment]
Screen Shot 2020-01-09 at 11 01 48 AM
Screen Shot 2020-01-09 at 11 02 59 AM
These overflows are not being applied properly

**Comment by salarenko** [comment]
Looks consistent on Chrome, Edge, IE11

**Comment by JKMarkowski** [comment]
Hi @stefanoScalzo, overflow is fixed now.

**Comment by JKMarkowski** [comment]
@InnaAtanasova Every comment is solved now

**Pull request merged**